src/sqlfluff/rules/L069.py

```diff
@@ -29,147 +35,366 @@ class Rule_L069(BaseRule):
+            else None
+        )
+
+        # If casting style is set to consistent,
+        # we use the casting style of the first segment we encounter.
+        if self.preferred_type_casting_style == "consistent":
+            memory = context.memory
+            prior_type_casting_style = context.memory.get("prior_type_casting_style")
+
+            # Construct fixes
+            if prior_type_casting_style == "cast":
+                if current_type_casting_style == "convert":
+                    # Get the content of CONVERT
+                    convert_content = self._get_children(
+                        FunctionalContext(context).segment.children(
+                            sp.is_type("bracketed")
                         )
                     )
-                )
+                    # We only care about 2-arguments convert
+                    # some dialects allow an optional 3rd argument e.g TSQL
+                    # which cannot be rewritten into CAST
+                    if len(convert_content) > 2:
+                        return None
 
-                # we only care about 2-arguments convert
-                # some dialects allow an optional 3rd argument e.g TSQL
-                # which cannot be rewritten into CAST
-                if len(convert_content) > 2:
-                    return None
-
-                # get the expression
-                expression = convert_content[1].raw
+                    fixes = self._cast_fix_list(
+                        context,
+                        convert_content[1],
+                        convert_content[0],
+                    )
+                elif current_type_casting_style == "shorthand":
+                    # Get the expression and the datatype segment
+                    expression_datatype_segment = self._get_children(
+                        FunctionalContext(context).segment
+                    )
 
-                # get the data type
-                datatype = convert_content[0].raw
+                    fixes = self._cast_fix_list(
+                        context,
+                        expression_datatype_segment[0],
+                        expression_datatype_segment[1],
+                        # We can have multiple shorthandcast e.g 1::int::text
+                        # in that case, we need to introduce nested CAST()
+                        expression_datatype_segment[2:],
+                    )
 
-                # Create fix to replace ``CONVERT`` with ``CAST``.
-                fix = LintFix.replace(
-                    context.segment,
-                    [
-                        CodeSegment(
-                            raw=f"CAST({expression} AS {datatype})",
+            elif prior_type_casting_style == "convert":
+                if current_type_casting_style == "cast":
+                    cast_content = self._get_children(
+                        FunctionalContext(context).segment.children(
```

Comment:
Nitpick: Maybe create the functional context once rather than repeating it several times, e.g.:
```
functional_context = FunctionalContext(context)
```